dev-build/make/files/make-4.4.1-r100-c23-fix.patch

```diff
@@ -0,0 +1,39 @@
+diff --git a/lib/fnmatch.c b/lib/fnmatch.c
+index 01da376..cb1c856 100644
+--- a/lib/fnmatch.c
++++ b/lib/fnmatch.c
+@@ -121,7 +121,7 @@ USA.  */
+    whose names are inconsistent.  */
+ 
+ # if !defined _LIBC && !defined getenv
+-extern char *getenv ();
++extern char *getenv (const char *);
+ # endif
+ 
+ # ifndef errno
+diff --git a/src/getopt.c b/src/getopt.c
+index 7a792de..76251cc 100644
+--- a/src/getopt.c
++++ b/src/getopt.c
+@@ -202,7 +202,7 @@ static char *posixly_correct;
+    whose names are inconsistent.  */
+ 
+ #ifndef getenv
+-extern char *getenv ();
++extern char *getenv (const char *);
+ #endif
+ 
+ static char *
+diff --git a/src/getopt.h b/src/getopt.h
+index df18cee..f84b36d 100644
+--- a/src/getopt.h
++++ b/src/getopt.h
+@@ -102,7 +102,7 @@ struct option
+    errors, only prototype getopt for the GNU C library.  */
+ extern int getopt (int argc, char *const *argv, const char *shortopts);
+ #else /* not __GNU_LIBRARY__ */
+-extern int getopt ();
++extern int getopt (int argc, char *const *argv, const char *shortopts);
```

Comment:
You're duplicating it with the line above. I think we're better off figuring out why its detection fails and reporting that upstream to gnulib once analysed (show me the report first). It is OK for it to not set `__GNU_LIBRARY__` for musl but we need it to get it right for musl in gnulib, rather than fixing every single consumer.

Reply:
I believe this was fixed long ago upstream in https://github.com/coreutils/gnulib/commit/1c64ecc467eb219cc70be133d3b17cc6eeea1369.